pkg/provider/infisical/api/api.go

```diff
@@ -22,16 +22,20 @@ import (
 	"net/http"
 	"net/url"
 	"strconv"
-	"time"
 
 	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
 	"github.com/external-secrets/external-secrets/pkg/metrics"
 	"github.com/external-secrets/external-secrets/pkg/provider/infisical/constants"
 )
 
+// HttpClient is an interface that wraps the Do method, so that it can be mocked out for testing.
+type HttpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
 type InfisicalClient struct {
 	BaseURL *url.URL
-	client  *http.Client
+	client  HttpClient
```

Comment:
Why is this needed?

Reply:
I mean you could just as easily create a test server with httptest package.

Reply:
No worries. Honestly, I'm a not 100% sure it's better, since it requires then a lot of fiddling with the server plus the setting up of the payloads and the responses... So..... this _might_ be better. :)

Reply:
It turned out to be just as clean (:

Reply:
Oh neat! :)